tools/dds/dds-server/lrs-device-controller.cpp

```diff
@@ -411,123 +512,85 @@ lrs_device_controller::~lrs_device_controller()
-    else
-    {
-        std::cout << "[\"";
-        size_t i = 0;
-        for( ; i < realdds_streams_to_start.size() - 1; ++i )
-            std::cout << realdds_streams_to_start[i].first << "\", \"";
-        std::cout << realdds_streams_to_start[i].first << "\"] streams started" << std::endl;
-    }
+    // We're here so all the profiles were acceptable; lock them in -- with no implicit profiles!
+    if( rsutils::json::get< bool >( msg, "commit", true ))
+        _bridge.commit();
 }
 
-void lrs_device_controller::stop_streaming( const json & msg )
-{
-    auto stream_names = msg["stream-names"];
 
-    size_t sensor_index = 0;
-    for ( std::string requested_stream_name : stream_names )
+std::vector< rs2::stream_profile >
+lrs_device_controller::get_rs2_profiles( realdds::dds_stream_profiles const & dds_profiles ) const
+{
+    std::vector< rs2::stream_profile > rs_profiles;
+    for( auto & dds_profile : dds_profiles )
     {
-        //Find sensor to close
-        if ( find_sensor( requested_stream_name, sensor_index ) )
-            break; //TODO - currently assumes all streams are from one sensor. Add support for multiple sensors
+        std::string stream_name = dds_profile->stream()->name();
+        std::string sensor_name = dds_profile->stream()->sensor_name();
 
-        if ( sensor_index == _sensors.size() )
-            throw std::runtime_error( "Could not find sensor to close `" + requested_stream_name + "`" );
+        auto it = _rs_sensors.find( sensor_name );
+        if( it == _rs_sensors.end() )
+        {
+            LOG_ERROR( "invalid sensor name '" << sensor_name << "'" );
+            continue;
+        }
+        auto & sensor = it->second;
+        auto rs2_profile = get_required_profile( sensor, stream_name, dds_profile );
+        rs_profiles.push_back( rs2_profile );
     }
-
-    //Stop streaming
-    _sensors[sensor_index].stop();
-    _sensors[sensor_index].close();
-    _dds_device_server->stop_streaming( stream_names );
-    std::cout << _sensors[sensor_index].get_info( RS2_CAMERA_INFO_NAME ) << " closed. Streams requested to stop: " << stream_names << std::endl;
+    return rs_profiles;
 }
 
+
 void lrs_device_controller::set_option( const std::shared_ptr< realdds::dds_option > & option, float new_value )
 {
-    size_t sensor_index = 0;
-    find_sensor( option->owner_name(), sensor_index );
-        
-    if( sensor_index == _sensors.size() )
-    {
-        //TODO - handle device options
-        throw std::runtime_error( "Could not find sensor with `" + option->owner_name() + "` stream" );
-    }
-
-    rs2_option opt_type = option_name_to_type( option->get_name(), _sensors[sensor_index] );
-
-    _sensors[sensor_index].set_option( opt_type, new_value );
+    auto it = _stream_name_to_server.find( option->owner_name() );
+    if( it == _stream_name_to_server.end() )
+        throw std::runtime_error( "no stream '" + option->owner_name() + "' in device" );
```

Comment:
Can we remove device level options?
They are not used and we don't implement to support them

Reply:
I think that's too much for this PR.
Let's do it separately...